### ISSUE TITLE
`no-useless-promise-resolve-reject` conflicts with `@typescript-eslint/return-await`

### ISSUE THREAD

**Issue by emmercm** [sindresorhus/eslint-plugin-unicorn#3503]
Here's a scenario where `no-useless-promise-resolve-reject` doesn't report an error, but I think it should:

```typescript
function foo(): string {
  return 'hello world';
}

async function bar(): Promise<string> {
  // If 'await' is removed, @typescript-eslint/return-await will add it back
  // If @typescript-eslint/return-await is ignored, then no-useless-promise-resolve-reject will remove the useless Promise.resolve()
  return await Promise.resolve(foo());
}
```

This is what I expect autofixes and editor suggestions (understanding that `@typescript-eslint/await-thenable` and `@typescript-eslint/require-await` may not autofix) to push me toward:

```typescript
function foo(): string {
  return 'hello world';
}

function bar(): string {
  return foo();
}
```
